internal/app/main.go

```diff
@@ -35,7 +35,20 @@ func Main() {
 		defer s.cleanup()
 	}
 
-	flags.readState(&s)
+	if err := flags.readState(&s); err != nil {
+		log.Fatal(err.Error())
+	}
+
+	if len(flags.target) > 0 && len(s.TargetMap) == 0 {
+		log.Info("No apps defined with -target flag were found, exiting")
```

Comment:
Should this be logged at `Error` or something?

Reply:
Also, this is kinda a weird check in the first place. It's checking if `s.readState()`, like, did what it's supposed to. It's almost a runtime test. If `flags` have targets and the state ends up not, it's a programming error we should cover in tests, not a runtime just-in-case kind of things. What do you think?

Reply:
This is the way it was before I've just moved this code but I think, in this case `helmsman` just exits with `0` saying "I have nothing to do so my work is done" and that's fine...

Reply:
Ah, so `s.TargetMap` is conditionally populated? e.g. `flags.target` can have targets and `s.TargetMap` can have none and that's ok?

Reply:
correct `s.TargetMap` is populated based on `flags.target` and/or `flags.group`